lang/java/compiler/src/test/java/org/apache/avro/compiler/specific/TestSpecificCompiler.java

```diff
@@ -927,6 +927,32 @@ public void testAdditionalToolsAreInjectedIntoTemplate() throws Exception {
   }
 
   @Test
+  public void testCamelize() {
+    assertEquals("snakeCase", SpecificCompiler.camelize("snake_case", false));
+    assertEquals("constantCase", SpecificCompiler.camelize("CONSTANT_CASE", false));
+    assertEquals("mixedSnake", SpecificCompiler.camelize("Mixed_Snake", false));
+    assertEquals("snakeUuid", SpecificCompiler.camelize("snake_UUID", false));
+    assertEquals("lower", SpecificCompiler.camelize("lower", false));
+    assertEquals("capitalized", SpecificCompiler.camelize("Capitalized", false));
+    assertEquals("caps", SpecificCompiler.camelize("CAPS", false));
+    assertEquals("camelCased", SpecificCompiler.camelize("camelCased", false));
+    assertEquals("camelCased", SpecificCompiler.camelize("CamelCased", false));
+    assertEquals("someUUID", SpecificCompiler.camelize("someUUID", false));
+    assertEquals("uuidFirst", SpecificCompiler.camelize("UUIDFirst", false));
+
+    assertEquals("SnakeCase", SpecificCompiler.camelize("snake_case", true));
+    assertEquals("ConstantCase", SpecificCompiler.camelize("CONSTANT_CASE", true));
+    assertEquals("MixedSnake", SpecificCompiler.camelize("Mixed_Snake", true));
+    assertEquals("SnakeUuid", SpecificCompiler.camelize("snake_UUID", true));
+    assertEquals("Lower", SpecificCompiler.camelize("lower", true));
+    assertEquals("Capitalized", SpecificCompiler.camelize("Capitalized", true));
+    assertEquals("CAPS", SpecificCompiler.camelize("CAPS", true));
```

Comment:
Is this intentional ? 
At https://github.com/apache/avro/pull/735/files#diff-bdda25ef6cc04f1ca556df625d9d4ecdd65951cfcd8b4e38da5916765bff75dbR946 `UUID` has been transformed to `Uuid` but here `CAPS` remains as `CAPS`. I'd expect `Caps` instead.